IviDriverCore/1.0/Spec/IviDriverCore.md

```diff
@@ -0,0 +1,770 @@
+
+## File Versioning
+
+IVI driver Windows DLLs shall contain a Windows version resource with the three entries: `CompanyName`, `ProductName`, and `FileVersion`.
+
+- `CompanyName` is a string that contains the value of the Driver Vendor or Driver Vendor property that the IVI driver returns.
+
+- `ProductName` is a string that includes the value of the Driver Identifier property that the IVI driver returns.
+
+- `FileVersion` is a string in the following format:
+  
+  `MajorVersion.MinorVersion.BuildVersion[.InternalVersion]`
+
+`MajorVersion`, `MinorVersion`, `BuildVersion`, and `InternalVersion` are decimal numbers greater than or equal to zero and less than or equal to 65535. The MajorVersion shall not be zero, except for a pre-release version of the initial release of a driver. Each number may contain leading zeros but may not exceed 5 digits. The number fields are separated by periods, with no embedded spaces. The maximum valid FileVersion is 65535.65535.65535.65535.
+
+A FileVersion string shall contain at least a `MajorVersion`, `MinorVersion`, and `BuildVersion`. A driver shall be released with an incremented `MajorVersion` or `MinorVersion` number when any of the following conditions are true:
+
+- The new version of the driver contains changes in its API syntax.
+
+- The new version of the driver contains significant changes to its semantics.
+
+A driver shall be released with at least an incremented `BuildVersion` number when any of the following conditions are true:
+
+- A new bitness of the driver is provided.
+
+- The driver is modified in a way that does not require a `MajorVersion` or `MinorVersion` update.
+
+If a driver vendor provides both a 32-bit and a 64-bit version of a driver, the 32-bit and 64-bit DLLs shall have the same `MajorVersion`, `MinorVersion`, and `BuildVersion`. Notice that when a 64-bit version of the driver is initially made available, the 32-bit version must be updated at the same time.
+
+For the purpose of determining the version of an IVI driver, only the `MajorVersion`, `MinorVersion`, and `BuildVersion` are used; the InternalVersion is not used. The InternalVersion is optional and reserved for use by driver suppliers.
+
+When comparing two `FileVersion` values, integer comparisons are performed successively for each number, starting at the leftmost number. If the first two numbers are equal, the next two numbers are compared, and so on.
+
+### Examples
+
+The following are examples of valid `FileVersion` values:
+
+    1.0.0
+    2.00.00
+    01.02.03
+    11.22.33
+    5.0.1
+    3.14.159
+    4.0.1000.0
+    0001.1.1.00005
+
+The following are examples of comparisons of `FileVersion` values for the purpose of comparing driver versions:
+
+    5.1.345.213 is greater than 4.3553.3244.234
+
+    5.1.345.213 is less than 5.1.346.213
+
+    5.2.13.26 is equal to 5.2.13.56
+
+    5.001.100.001 is equal to 5.1.00100.1
+
+    2.15.32.1 is greater than 2.15.18.1
+
+    1.1.1 is equal to 1.1.1.0
```

Comment:
I recommend adding an example that makes it clear that these aren't to be treated like decimal numbers:

    1.1.2 is less than 1.1.10